qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/ReaderCloseTest.java

```diff
@@ -70,6 +70,14 @@ public void shouldDeleteCompactedSegmentsFilesAfterLeaderChange() throws IOExcep
             .getCluster()
             .getNodeId();
     clusteringRule.forceClusterToHaveNewLeader(followerId);
+    // because of https://github.com/camunda-cloud/zeebe/issues/8329
+    // we need to add another record so we can do a snapshot
+    clientRule
+        .getClient()
+        .newPublishMessageCommand()
+        .messageName("test")
+        .correlationKey("test")
+        .send();
```

Comment:
This is just a workaround for now ideally we fix https://github.com/camunda-cloud/zeebe/issues/8329

Reply:
Please correct me if I'm wrong. #8329 exist as a problem as well without the changes in this PR, but this PR makes the impact of the problem larger. If that's the case, then perhaps we shouldn't merge this PR until we've fixed this bug. 

❓ Is this bug blocking this PR in your opinion?

Reply:
You're right that it the change increases the impact, but still the likelihood is really low. This can only happen if no event is appended from outside or the processor doesn't process anything, otherwise it will update the commit position and then it is all fine. I think @npepinpe can decide here.

Reply:
From #8329, it sounds to me like the impact is minimal? You won't be able to take snapshot if nothing is committed, but chances are if you're processing anything, some follow up event will be committed. If you're replaying, it just delays until the next entry is committed, which greatly lowers the potential impact: either nothing is committed (meaning nothing was replicated, meaning disk does not grow either), or something is and a snapshot can be taken. It would only affect the case where you're only exporting but nothing is to be processed, which is not very likely. I think we can go ahead and merge, unless I'm missing something.